client/state/routing/middleware.js

```diff
@@ -9,11 +9,18 @@ import page from 'page';
  */
 import localforage from 'lib/localforage';
 import { isOutsideCalypso } from 'lib/url';
-import { ROUTE_SET } from 'state/action-types';
+import { ROUTE_SET, ROUTE_TO } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
 const LAST_PATH = 'last_path';
 
+export const routingMiddleware = () => ( next ) => ( action ) => {
+	if ( action.type === ROUTE_TO ) {
+		page( action.path );
+	}
```

Comment:
> Adding this routing middleware is one of those "small tasks" that's been on my list for some time now but I haven't gotten around to it.

I was a bit surprised it didn't exist already 😄 

> Ultimately I think we'll have the best option if we have a decoder/encoder pair for the section.

I like the feel of those functions, though I'll leave that for the next PR.